pyboy/plugins/game_wrapper_pokemon_gen1.py

```diff
@@ -28,6 +45,24 @@ class GameWrapperPokemonGen1(PyBoyGameWrapper):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, game_area_section=(0, 0, 20, 18), game_area_follow_scxy=True, **kwargs)
         self.sprite_offset = 0
+        if self.pyboy.memory[ADDR_TITLE_SCREEN] == 0:
+            pass
+        else:
+            self.number_of_pokemon = self.pyboy.memory[ADDR_NUMBER_OF_POKEMON]
+            self.total_items = self.pyboy.memory[ADDR_TOTAL_ITEMS]
+            # Extract the time in game:
+            _hours = self.pyboy.memory[ADDR_HOURS]
+            _minutes = self.pyboy.memory[ADDR_MINUTES]
+            _seconds = self.pyboy.memory[ADDR_SECONDS]
+            self.game_time = f'{_hours}:{_minutes}:{_seconds}'
+
+            # Check whether the player is in battle mode or out of battle
+            if self.pyboy.memory[ADDR_BATTLE] == 1:
+                self.battle_state = 'In Battle'
+            else:
+                self.battle_state = 'Overworld'
+
+            self.badges = self.pyboy.memory[ADDR_BADGES] #Research how this works.
```

Comment:
This is all duplicate code. Better just define some default value (0) for all of them. The data of the game wrapper classes are not valid before calling `pyboy.tick()` anyway.